app/models/bike.rb

```diff
@@ -161,36 +161,69 @@ def organization(org_or_org_id)
       includes(:bike_organizations).where(bike_organizations: { organization_id: organization.id })
     end
 
-    # Possibly-found bikes are stolen bikes that have a counterpart record
+    # Possibly-found bikes are stolen bikes that have a counterpart record(s)
     # (matching by normalized serial number) in an abandoned state.
     def possibly_found
-      abandoned_serials =
-        abandoned
-          .where.not(stolen: true)
-          .select(:serial_normalized)
-
       unscoped
         .current
         .stolen
-        .where.not(abandoned: true)
-        .where(serial_normalized: abandoned_serials)
+        .non_abandoned
+        .where(serial_normalized: abandoned.non_stolen.select(:serial_normalized))
```

Comment:
I think it would be better to `pluck` here

Reply:
What would Benny Wong say? 🤔 
(I'm forced to guess at / interpolate context and reasoning here.)

Contra `pluck`: Because `select` evaluates to an AR relation, using it here triggers a single query overall. `pluck` triggers two, and is slower:

```rb
Bike.possibly_found
# (198.0ms)  SELECT COUNT(*) FROM "bikes" WHERE "bikes"."example" = $1 AND "bikes"."hidden" = $2 AND "bikes"."deleted_at" IS NULL AND "bikes"."stolen" = $3 AND "bikes"."abandoned" = $4 AND "bikes"."serial_normalized" IN (SELECT "bikes"."serial_normalized" FROM "bikes" WHERE "bikes"."example" = $5 AND "bikes"."hidden" = $6 AND "bikes"."deleted_at" IS NULL AND "bikes"."abandoned" = $7 AND "bikes"."stolen" = $8  ORDER BY listing_order desc)  [["example", "f"], ["hidden", "f"], ["stolen", "t"], ["abandoned", "f"], ["example", "f"], ["hidden", "f"], ["abandoned", "t"], ["stolen", "f"]]
# Bike Load (130.2ms)  SELECT "bikes".* FROM "bikes" WHERE "bikes"."example" = $1 AND "bikes"."hidden" = $2 AND "bikes"."deleted_at" IS NULL AND "bikes"."stolen" = $3 AND "bikes"."abandoned" = $4 AND "bikes"."serial_normalized" IN (SELECT "bikes"."serial_normalized" FROM "bikes" WHERE "bikes"."example" = $5 AND "bikes"."hidden" = $6 AND "bikes"."deleted_at" IS NULL AND "bikes"."abandoned" = $7 AND "bikes"."stolen" = $8  ORDER BY listing_order desc)  [["example", "f"], ["hidden", "f"], ["stolen", "t"], ["abandoned", "f"], ["example", "f"], ["hidden", "f"], ["abandoned", "t"], ["stolen", "f"]]

Bike.possibly_found_with_pluck
# (52.1ms)  SELECT "bikes"."serial_normalized" FROM "bikes" LEFT OUTER JOIN "colors" ON "colors"."id" = "bikes"."tertiary_frame_color_id" LEFT OUTER JOIN "colors" "secondary_frame_colors_bikes" ON "secondary_frame_colors_bikes"."id" = "bikes"."secondary_frame_color_id" LEFT OUTER JOIN "colors" "primary_frame_colors_bikes" ON "primary_frame_colors_bikes"."id" = "bikes"."primary_frame_color_id" LEFT OUTER JOIN "stolen_records" ON "stolen_records"."id" = "bikes"."current_stolen_record_id" AND "stolen_records"."current" = $1 WHERE "bikes"."example" = $2 AND "bikes"."hidden" = $3 AND "bikes"."deleted_at" IS NULL AND "bikes"."abandoned" = $4 AND "bikes"."stolen" = $5  ORDER BY listing_order desc  [["current", "t"], ["example", "f"], ["hidden", "f"], ["abandoned", "t"], ["stolen", "f"]]
# (57.2ms)  SELECT COUNT(*) FROM "bikes" WHERE "bikes"."example" = $1 AND "bikes"."hidden" = $2 AND "bikes"."deleted_at" IS NULL AND "bikes"."stolen" = $3 AND "bikes"."abandoned" = $4 AND ("bikes"."serial_normalized" IN ('5NF5D12HG1615', 'F N GD6D255', '5520518', 'CM150520146', 'CM13G515132', '2595T0087D', 'P3DC26547', '1M33513', '6009619', '14H25430', 'U90421804', 'AJ30018231', '8243035130', 'WTU0181105', '5NF 10FP1387', 'W8K255322T', '1080600585', 'M15061152', 'F031102774F25', 'H819DD1158 FR80140', 'F96J98812', 'P2JA01627', '5NF3D16A86926', 'AAA18F00844', '81 12189576', 'W13435191', 'E3727910', '16A25772', 'CD515347017', 'W58C601025554K', '80C52222361U0419', '1CF6107761', 'AC16F11135', 'F002K20486', '1CC011F00315', 'X22997', 'P9EA611855', '5NX9514JA01264', 'F3P8J0806', 'HA8083061595C5N18') OR "bikes"."serial_normalized" IS NULL)  [["example", "f"], ["hidden", "f"], ["stolen", "t"], ["abandoned", "f"]]
# Bike Load (145.3ms)  SELECT "bikes".* FROM "bikes" WHERE "bikes"."example" = $1 AND "bikes"."hidden" = $2 AND "bikes"."deleted_at" IS NULL AND "bikes"."stolen" = $3 AND "bikes"."abandoned" = $4 AND ("bikes"."serial_normalized" IN ('5NF5D12HG1615', 'F N GD6D255', '5520518', 'CM150520146', 'CM13G515132', '2595T0087D', 'P3DC26547', '1M33513', '6009619', '14H25430', 'U90421804', 'AJ30018231', '8243035130', 'WTU0181105', '5NF 10FP1387', 'W8K255322T', '1080600585', 'M15061152', 'F031102774F25', 'H819DD1158 FR80140', 'F96J98812', 'P2JA01627', '5NF3D16A86926', 'AAA18F00844', '81 12189576', 'W13435191', 'E3727910', '16A25772', 'CD515347017', 'W58C601025554K', '80C52222361U0419', '1CF6107761', 'AC16F11135', 'F002K20486', '1CC011F00315', 'X22997', 'P9EA611855', '5NX9514JA01264', 'F3P8J0806', 'HA8083061595C5N18') OR "bikes"."serial_normalized" IS NULL)  [["example", "f"], ["hidden", "f"], ["stolen", "t"], ["abandoned", "f"]]
```

```rb
timeit { Bike.possibly_found }
# => 0.0008

timeit { Bike.possibly_found_with_pluck }
# => 0.0744

timeit { Bike.possibly_found.to_a }
# => 0.1422

timeit { Bike.possibly_found_with_pluck.to_a }
# => 0.7154
```

Reply:
OH SNAP. Cool, I had no idea it evaluated to a single query, that's neat.